stdlib/public/core/Collection.swift

```diff
@@ -646,6 +646,7 @@ public protocol Collection : _Indexable, Sequence {
   /// supplying `IndexingIterator` as its associated `Iterator`
   /// type.
   associatedtype Iterator = IndexingIterator<Self>
+    where Self.Iterator.Element == _Element
```

Comment:
Could we rename _Element to Element, and write Element in place of Iterator.Element everywhere? :-)